zummon/src/commonMain/kotlin/companysurvey/Validation.kt

```diff
@@ -339,25 +339,69 @@ class ElectricityValidator : Validator<Electricity> {
-            ValidationResult(Status.VALID, translate("electricity.quarterHourlyDeliveryLowContractedCapacityKw", contractedCapacity))
+        val contractedCapacity_kW = electricity.getContractedConnectionCapacityKw()
+        if (contractedCapacity_kW == null) {
+            return ValidationResult(Status.MISSING_DATA, message(
+                en = "Gecontracteerd vermogen levering ontbreek",
+                nl = "Contracted delivery capacity missing",
+            ))
+        }
+
+        if (electricity.quarterHourlyDelivery_kWh == null) {
+            return ValidationResult(Status.MISSING_DATA, message(
+                en = "Kwartierwaarden levering ontbreek",
+                nl = "Quarter-hourly delivery missing",
+            ))
+        }
+
+        val peakDelivery = electricity.quarterHourlyDelivery_kWh.getPeak()
+
+        return if ( peakDelivery.kW() <= contractedCapacity_kW) {
+            ValidationResult(Status.VALID, message(
+                en = "Piek van kwartierwaarden levering ${peakDelivery.kWh()} kWh valt binnen gecontracteerd vermogen levering ${contractedCapacity_kW} kW",
+                nl = "Peak of quarter-hourly delivery ${peakDelivery.kWh()} kWh does not exceed contracted capacity ${contractedCapacity_kW} kW",
+            ))
         } else {
-            ValidationResult(Status.INVALID, translate("electricity.quarterHourlyDeliveryHighContractedCapacityKw", contractedCapacity))
+            ValidationResult(Status.INVALID, message(
+                nl = "Piek van kwartierwaarden levering ${peakDelivery.kWh()} kWh mag niet hoger zijn dan gecontracteerd vermogen levering $contractedCapacity_kW kW",
+                en = "Peak of quarter-hourly delivery ${peakDelivery.kWh()} kWh should be below contracted capacity $contractedCapacity_kW Kw",
+            ))
         }
     }
 
     //peak of feed-in should be less than contracted capacity
     fun quarterFeedInLowContractedCapacity(electricity: Electricity): ValidationResult {
-        val contractedFeedInCapacity = (electricity.getContractedFeedInCapacityKw() ?: 0.0).toFloat()
-        val pickFeedIn = electricity.quarterHourlyFeedIn_kWh?.values?.maxOrNull() ?: Float.MIN_VALUE
-        return if (pickFeedIn < contractedFeedInCapacity) {
-            ValidationResult(Status.VALID, translate("electricity.quarterHourlyDeliveryLowContractedCapacityKw", contractedFeedInCapacity))
+        val contractedFeedInCapacity_kW = (electricity.getContractedFeedInCapacityKw() ?: 0.0).toFloat()
+        if (contractedFeedInCapacity_kW == null) {
+            return ValidationResult(Status.MISSING_DATA, message(
+                en = "Gecontracteerd vermogen teruglevering ontbreek",
+                nl = "Contracted feed-in capacity missing",
+            ))
+        }
+
+        if (electricity.quarterHourlyFeedIn_kWh == null) {
+            // Or not applicable if no solar panels
+            return ValidationResult(Status.MISSING_DATA, message(
```

Comment:
Message added in validateAnnualFeedInMatchesQuarterHourlyFeedIn

Reply:
I've changed it from MISSING_DATA to NOT_APPLICABLE so there is no duplicate message.